arch/cortex-m/src/scb.rs

```diff
@@ -54,3 +54,13 @@ pub unsafe fn reset() {
     let reset = (0x5FA << 16) | (aircr & (0x7 << 8)) | (1 << 2);
     SCB.aircr.set(reset);
 }
+
+/// Enables Floating Point Unit. Section 4.6.6.
+/// http://infocenter.arm.com/help/index.jsp?topic=/com.arm.doc.dui0553a/BEHBJHIG.html
+/// http://infocenter.arm.com/help/index.jsp?topic=/com.arm.doc.dai0298a/CEGJEIGH.html
+pub unsafe fn enable_fpu() {
```

Comment:
Is there a hardware register we can read to validate that a FPU is present? I would prefer this throw an exception / error / etc if there isn't an FPU

Reply:
what about feature gating it at compile time? So it just wouldn't be available to call on a non-floating point enabled MCU?

Reply:
This is certainly should be feature-gated since there's no way to check FPU presence using hardware registers/flags.

Reply:
That'd be fine too, though there's probably a bigger architectural discussion we need to have surrounding how to manage such `cfg`s for varying chips